---
affilliation: TUFTS UNIVERSITY MEDFORD
description: 'Each human is, on average, colonized by 1014 microbial cells that mostly
  reside in the gastrointestinal track. Research in the last two decades has uncovered
  the central role of this microbial community in human health and disease. A pressing
  challenge, however, is the lack of understanding of microbial drug metabolism. Experimental
  studies, clinical observations, and anecdotal examples demonstrate that microbial
  enzymes alter drugs through common enzymatic transformations such as reduction,
  hydrolysis, dehydroxylation, demethylation, and others. Despite progress, there
  lacks a systematic approach for the discovery and analysis for such transformations,
  thus hindering the design and interpretation of experimental studies. There is therefore
  a need to establish workflows to explore such transformations. We investigate in
  this proposal microbial drug metabolism at the molecular and community levels. We
  are proposing to use data from two Common Fund data sets to conduct this investigation.
  Illuminating the Druggable Genome (IDG) catalogues drugs and their pharmacologic
  action, while the NIH Human Microbiome Project (HMP) provides detailed gut microbial
  data for cohorts. We are also proposing to use our deep-learning tools to predict
  the likelihood of interaction between microbial enzymes and drugs (Aim 1), and to
  predict putative derivative products due to this interaction (Aim 2). Our tools
  (CSI for Aim 1, and GNN-SOM and PROXIMAL for Aim 2) have already been validated
  on other datasets and in other studies, and they will be adapted for microbial enzymes
  and drugs based on data culled from IDG and HMP and other resources. The workflows
  established in Aims 1 and 2 will be utilized to conduct a pilot study (Aim 3) to
  investigate the extent of functional redundancy towards drugs within microbial communities
  of healthy individuals that are culled from HMP. The strength of our Approach therefore
  lies in: i) adapting novel, state-of-the-art deep-learning models to predict microbial
  enzyme promiscuity on drugs, ii) providing biochemically explainable drug products,
  and iii) exploring how drug microbial metabolism is a function of microbial community
  composition. The Significance of this research is that it provides an explainable
  hypothesis of microbial drug metabolism. The work is impactful as it will enable
  further studies, such as exploring the functional redundancy of a microbial community
  towards drugs (as planned in Aim 3) and designing and interpreting experimental
  studies involving the impact of the gut microbiota on drugs. The proposed work is
  appropriate for this funding opportunity as it curates and annotates data using
  novel deep-learning approaches and creates a previously unexplored link between
  the HMP and IDG. '
end_date: '2025-09-19T12:00:00-04:00'
grant_num: R03OD036490
pi: HASSOUN, SOHA
title: Using Common Fund Datasets to Illuminate Drug-Microbial Interactions
---