---
affilliation: YALE UNIVERSITY
end_date: '2023-08-13T12:00:00-04:00'
grant_num: R03OD032623
pi: XU, JINRUI
title: Using three-dimensional genome structure to refine eQTL detection
---
The Genotype-Tissue Expression (GTEx) Program studies the impact of genetic variants on gene expression in many human cell types and tissues. To identify the expression quantitative trait loci (eQTLs) of each gene, the genetic variants within one million base pairs (1 Mb) of the transcription start site (TSS) of the gene are considered as the candidates, and then the GTEx computational pipeline identifies the significant candidates as eQTLs of the gene. This 1 Mb threshold is being widely used as the gold standard in the field to reduce multiple tests. Using this threshold assumes that genetic variants outside of this distance contribute little to gene expression, and thus are unlikely to be eQTLs. However, we observed that, on average, 10% of cis-regulatory elements (CREs) are outside of the 1 Mb threshold, herein referred to as distal CREs. Therefore, the eQTLs in such CREs are missed using the 1 Mb threshold. In addition, the 1 Mb threshold implicitly assumes that the majority of genomic regions within the distance to a TSS are CREs that regulate the gene. However, we found that on average CREs account for only 2.1% of the Â±1Mb regions around a TSS. Moreover, it is not uncommon that CREs skip the closest genes to regulate distal genes. These observations indicate that many candidate variants within the 1 Mb distance may be noise, and thus impede the detection of bona fide eQTLs. In line with this, we found that using distance thresholds smaller than 1 Mb substantially increase the numbers of eQTLs and associated genes. These results together indicate that the current eQTLs detection can be improved by focusing only on the CREs of genes. To this end, we will use the genome structure data from 4D Nucleome and other public data to build CRE-gene linkages. These linkages are expected to detect more eQTLs, especially the weak ones. The results will enhance the existing GTEx dataset and substantially improve our understanding of gene expression regulation and human diseases.