---
affilliation: UNIVERSITY OF CALIFORNIA AT DAVIS
description: "Prof. Oliver Fiehn will work with his key persons, statistician Dr.\
  \ Christopher Brydges, bioinformatics specialist Dr. Yuanyue Li and programmer Gert\
  \ Wohlgemuth (all UC Davis) to generate new pipelines that integrate stool microbial\
  \ metagenomics data and stool mass spectrometry data to better associate metabolites\
  \ with disease progression in inflammatory bowel disease. We will work in consultation\
  \ with Dr. Clary Clish (Broad Institute) who generated and deposited the data to\
  \ the NIH Common Funds MetabolomicsWorkbench and the iHMP integrated human microbiome\
  \ data. We will prioritize the enormous set of more than 80,000 yet unidentified\
  \ stool metabolic signals using longitudinal disease progression over 1 year in\
  \ subjects with inflammatory bowel disease, in comparison to healthy subjects. For\
  \ this limited set of not more than 1,000 metabolites that will show significant\
  \ association with health outcomes, we will use all available accurate mass MS/MS\
  \ data and all stool microbiome data to obtain metabolite class information and\
  \ likely metabolite structures or substructures. Dr. Clish will review our results\
  \ and share new annotations that his group will release. To this end, we will develop\
  \ the tools for metabolome predictions that have been built by the KBase collaborative\
  \ research consortium over the past 10 years. KBase uses microbial genomic sequences\
  \ (or even transcriptomics data) to automatically build metabolic pathways through\
  \ enzyme predictions and gap filling. KBase also empowers utilization of microbial\
  \ communities, modeling import and export of metabolites that other microbes can\
  \ use as carbon sources. In consultation with Dr. Chris Henry (Argonne National\
  \ Lab) from the KBase consortium, we will then build pipelines within the KBase\
  \ environment to include mass spectrometry tools that the Fiehn laboratory has built\
  \ through its past NIH funding, specifically formula predictions and substructure\
  \ predictions (in MS-FINDER), retention time predictions (in Retip.app), hybrid-shift\
  \ MS/MS similarity matching (in NIST search), and entropy similarity MS/MS matching\
  \ (in MassBank.us). This specific project will have large impact on other, similar\
  \ microbiome/metabolome projects that will be uploaded to the NIH Common Funds databases\
  \ in the future. The project addresses the huge complexity in stool metagenomics\
  \ and stool metabolomics data, and delivers key pipelines (called \u2018narratives\u2019\
  \ in KBase) that can be used by the research community at large."
end_date: '2024-09-19T12:00:00-04:00'
grant_num: R03OD034497
pi: FIEHN, OLIVER
title: Integrating metagenomics data into accurate mass stool metabolite identifications
---