---
affilliation: WEILL MEDICAL COLL OF CORNELL UNIV
description: 'Pediatric brain tumors are the most frequent cause of morbidity in children
  with solid tumors. Importantly, the aggressive therapeutic regiments often lead
  to debilitating neurological effects. The realization that developmental processes
  critical to brain development are also deregulated in cancer has provided new hope
  for understanding and treating brain tumors. Indeed, single cell-RNAseq analyses
  have further demonstrated the role of defects in lineage determination for pediatric
  brain tumors. To discover novel drivers of tumorigenesis, we will focus on the function
  of three-dimensional (3D) genome folding in pediatric brain tumors. Indeed, 3D chromatin
  interactions are involved in gene expression regulation, and changes in genome folding
  are linked to cell identity acquisition during development. While there is increasing
  interest in elucidating the function of 3D genome architecture during developmental
  processes and in cancer, how the 3D genome is organized in different pediatric brain
  tumors and its roles in tumor formation and progression are unknown. We hypothesize
  that disrupted 3D genome folding during embryonic or postnatal development alters
  gene expression leading to abnormal cell differentiation and tumorigenesis in the
  developing brain. To test our hypothesis, we will comprehensively interrogate the
  genomes of pediatric brain tumors for non-coding variants that may affect 3D genome
  folding. We will use a deep-learning model called Akita that predicts 3D chromatin
  interaction frequencies from genome sequence alone. Because Akita only requires
  DNA sequence as input, we can predict the effect of any variant within a single
  framework that accommodates single-nucleotide variants (SNVs), insertion/deletions
  (indels), and structural variation (SVs). Akita will be used with pediatric brain
  whole genome sequences (WGS) from Gabriella Miller Kids First (KF) plus chromatin
  capture, epigenetic, and expression data from the 4D Nucleome (4DN) and Genotype-Tissue
  Expression (GTEx) programs in the following aims: 1) Determine the 3D genome architecture
  of Atypical teratoid/rhabdoid tumor AT/RT tumors. We have initiated our study using
  AT/RT, tumors thought to be due to defects in early development11 and the most common
  brain tumor in children less than six months of age. 1.A. We will develop a bioinformatics
  pipeline that uses Akita to quantify how much a genetic variant is predicted to
  disrupt 3D chromatin interactions in AT/RT tumors. 1.B. We will validate and determine
  the functional relevance of 3D genomic folding disruptions observed in AT/RT tumors.
  2) Determine the 3D genome architecture of malignant pediatric tumors. We will extend
  our analyses with Akita to additional malignant pediatric brain tumors, focusing
  for this pilot project on the most malignant and treatment refractory tumors. This
  innovative project, using a new deep-learning tool Akita, will lead to, novel research
  hypotheses and will accelerate the discovery of additional regulators of pediatric
  cancer tumorigenesis and thus to potential therapeutic strategies for these devastating
  diseases.'
end_date: '2024-09-19T12:00:00-04:00'
grant_num: R03OD034499
pi: RESNICK, ADAM CAIN, DORRESTEIN, PIETER C (contact)
title: Deciphering the 3D genome of pediatric brain tumors
website: https://github.com/ketringjoni/SuPreMo
---